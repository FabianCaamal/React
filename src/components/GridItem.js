import React from 'react'
import PropTypes from 'prop-types'

export const GridItem = ({ title, url }) => {
    return (
        <div className='card animate__animated animate__fadeInDown' >
            <img src={ url } alt={ title } />
            <p> { title } </p>
        </div>
    )
}

GridItem.propType = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
