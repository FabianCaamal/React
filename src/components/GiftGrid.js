import React from 'react'
import { useFetchGifts } from '../hooks/useFetchGifts'
import { GridItem } from './GridItem';

export const GiftGrid = ({ category }) => {

    const { data:images , loading } = useFetchGifts(category);

    return (
        <>
            <h3> { category } </h3>

            { loading && <p>Cargando...</p> }

            <div className='card-grid'>

                {
                    images.map( img => (
                        <GridItem key={img.id} {...img} />
                    ))
                }

            </div>
            <hr/>
        </>
    )
}
