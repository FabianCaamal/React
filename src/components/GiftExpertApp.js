import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GiftGrid } from './GiftGrid';

export const GiftExpertApp = () => {

    const [categories, setCategories] = useState(['one punch man']);

    // const handleAdd = () => {
    //     setCategories([...categories,'Naruto']);
    // }

    return (
        <>
            <h2>Buscador de gifts</h2>
            <AddCategory setCategories={setCategories} />
            <hr/>

            <ol>
                {
                    categories.map( cat => (
                        <GiftGrid key={cat} category={cat} />
                    ))
                }
            </ol>
        </>
    )
}
    
