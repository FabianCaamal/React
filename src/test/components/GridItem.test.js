import React from 'react';
import {shallow} from 'enzyme';
import { GridItem } from "../../components/GridItem";

describe('Pruebas de GridItem', () => {

    const title = 'titulo';
    const url = 'https://google.com';

    const wrapper = shallow(<GridItem title={title} url={url} />);

    test('Debe mostrar el campo correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('debe tener un parrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect(p.text().trim() ).toBe(title);
    });

    test('debe teener imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe tener la clase: animate__fadeInDown', () => {
        const div = wrapper.find('div');

        const className = div.prop('className');
        
        expect(className.includes('animate__fadeInDown')).toBe(true);
    });
    
    
});
