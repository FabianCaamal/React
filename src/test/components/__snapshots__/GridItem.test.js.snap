// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GridItem Debe mostrar el campo correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="titulo"
    src="https://google.com"
  />
  <p>
     
    titulo
     
  </p>
</div>
`;
